app/scripts/lib/stores/persistence-manager.ts

```diff
@@ -1,9 +1,49 @@
+/**
+ * navigator.locks is a polyfill for the Locks API, which can be removed from
+ * this file after we drop support for older Firefox versions.
+ * navigator.locks is supported in Firefox 102+.
+ */
+import 'navigator.locks';
 import log from 'loglevel';
 import { captureException } from '@sentry/browser';
 import { isEmpty } from 'lodash';
-import { type MetaMaskStateType, MetaMaskStorageStructure } from './base-store';
-import ExtensionStore from './extension-store';
-import ReadOnlyNetworkStore from './read-only-network-store';
+import { KeyringControllerState } from '@metamask/keyring-controller';
+import { MISSING_VAULT_ERROR } from '../../../../shared/constants/errors';
+import { OnboardingControllerState } from '../../controllers/onboarding';
+import { PreferencesControllerState } from '../../controllers/preferences-controller';
+import { AppStateControllerState } from '../../controllers/app-state-controller';
+import { IndexedDBStore } from './indexeddb-store';
+import type {
+  MetaMaskStateType,
+  MetaMaskStorageStructure,
+  BaseStore,
+  MetaData,
+} from './base-store';
+
+export type Backup = {
+  KeyringController?: KeyringControllerState;
+  OnboardingController?: OnboardingControllerState;
+  PreferencesController?: PreferencesControllerState;
+  AppStateController?: AppStateControllerState;
+};
+
+function makeBackup(state?: MetaMaskStateType): Backup {
+  return {
+    KeyringController: state?.KeyringController,
+    OnboardingController: state?.OnboardingController,
+    PreferencesController: state?.PreferencesController,
```

Comment:
Which properties specifically were needed from `PreferencesController`? I did expect it to be required here, but it would be good to have a record of why. This controller is a bit of a junk drawer, has a lot of stuff we don't strictly need, so maybe would be a candidate for splitting up in the future.

Reply:
I actually didn't look into the reasons for backing up these other properties. I did try only the `vault`, as you saw... but that alone couldn't be used to recover the database without refactoring the unlock and onboarding flows.

Should we go through all the bits of state together to decide what to keep?